### PR TITLE
Check if user's browser is supported

### PR DESCRIPTION
**Pull request by GregoryHorvath**
Ref.: https://github.com/IFRCGo/go-frontend/issues/1548

Not entirely sure if I put all this in the right place, and if I used `useMemo` right (it's for something like caching values?)

## Changes
- Added https://www.npmjs.com/package/browser-detect?activeTab=readme
- The `Root` component's content is only rendered if the browser is not IE or Edge (probably this will be adjusted)

### CODE REVIEW

File: src/root/index.js [hunk 1]
```diff
@@ -10,22 +11,37 @@ import Multiplexer from './Multiplexer';
 
 require('isomorphic-fetch');
 
+const browserNames = {
+  ie: 'Internet Explorer',
+  edge: 'Edge',
+  firefox: 'Firefox',
+  chrome: 'Chrome'
+};
+const unsupportedBrowsers = ['ie', 'edge'];
```
Comment (batpad) [inline src/root/index.js:20]: This should only be IE and not Edge?

**Comment by batpad** [comment]
@GregoryHorvath it is likely a better idea to not load the React component at all, since even that may not work in IE - so @frozenhelium suggested putting it in the root index.js here: https://github.com/IFRCGo/go-frontend/blob/develop/src/index.js -- so here we check the browser, and just use a plain document.getElementById('app-container').innerHTML = '' to set the HTML and not React methods. Let me know if that makes sense, cc @frozenhelium

File: src/root/index.js (hunk 1 above)
Comment (GregoryHorvath) [reply src/root/index.js]: https://github.com/IFRCGo/go-frontend/pull/1566/commits/f953f2436c7116616e00271d1b49a862a941b12a moved the handling to [/src/index.js](https://github.com/IFRCGo/go-frontend/blob/develop/src/index.js) only including IE now

**Comment by batpad** [comment]
This looks good to me @GregoryHorvath - do we have a good way to test this actually works in Internet Explorer?

**Comment by GregoryHorvath** [comment]
@batpad Good question, just deployed it to surge (database is not set, doesn't matter here): https://ifrc-go-feature-browser-detect.surge.sh

But IE almost crashes under it and still just showing the grey screen. I'll have to look into this more.

**Comment by GregoryHorvath** [comment]
@batpad Well, I'm somewhat stuck with it with no idea where to go next.

It gives 2 errors:
image

One is pointing to an => function in `/node_modules/ansi-styles/index.js`, and one is to the `Multiplexer`
```javascript
const wrapAnsi16 = (fn, offset) => function () {
	const code = fn.apply(colorConvert, arguments);
	return `\u001B[${code + offset}m`;
};

---------------------------------------------

function Multiplexer(props) {
  const {
    getUser,
```

I've tried:
1. adding IE to the `targets` in Babel config
2. adding `import 'react-app-polyfill/ie11';`
3. removing all other imports from `/src/index.js` than `import browser from 'browser-detect';`

Even with 3. it gave me the error with the => function.

**Comment by GregoryHorvath** [comment]
Closing this, we don't want to use a 3rd party package for this, simple JS should do.

**Pull request closed without merging**